cicd/jenkins/jenkins-master/docker/Dockerfile

```diff
@@ -1,21 +1,103 @@
-FROM jenkins/jenkins:2.263.3-lts
+FROM jenkins/jenkins:2.277.1-lts
 
 USER root
 
+# We have to downgrade libssl to u4 from u5 due to libssl-dev depending on u5
+# Will want to track this and remove in the future.
 RUN echo "deb http://ftp.debian.org/debian stable main contrib non-free" > /etc/apt/sources.list && \
     apt-get update -qq && \
-    apt-get install -qqy apt-transport-https ca-certificates curl gnupg2 software-properties-common gir1.2-glib-2.0 gir1.2-packagekitglib-1.0 python-apt-common python3 python3-venv python3-dbus python3-gi python python-pip python3-software-properties python3-pip python3-wheel python3-setuptools ruby ruby-bundler sudo less vim-tiny gettext-base jq
+    apt-get install -qqy --allow-downgrades --reinstall libssl1.1=1.1.1d-0+deb10u4 && \
+    apt-get install -qqy apt-transport-https ca-certificates curl gcc gnupg2 \
+    gir1.2-glib-2.0 gir1.2-packagekitglib-1.0 gettext-base	jq less make\
+    netbase python-apt-common ruby ruby-bundler software-properties-common sudo \
+    tzdata vim-tiny wget
 
-RUN pip  install --upgrade pip wheel setuptools && \
-    curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -
+# Setup python stdlib dependencies
+RUN set -ex \
+    && apt-get update && apt-get install -y --no-install-recommends \
+        dpkg-dev \
+        libbluetooth-dev \
+        libbz2-dev \
+        libc6-dev \
+        libexpat1-dev \
+        libffi-dev \
+        libgdbm-dev \
+        liblzma-dev \
+        libncursesw5-dev \
+        libreadline-dev \
+        libsqlite3-dev \
+        libssl-dev \
+        tk-dev \
+        uuid-dev \
+        xz-utils \
+        zlib1g-dev;
 
-RUN add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/debian $(lsb_release -cs) stable" && \
-    apt-get update  -qq && \
-    apt-cache madison docker-ce && \
-    apt-get install docker-ce=5:19.03.12~3-0~debian-stretch docker-ce-cli=5:19.03.12~3-0~debian-stretch containerd.io -y && \
+# Install python3.x
+ENV PYTHON_INSTALL_VERSION 3.9.2
+RUN wget -O python.tar.xz "https://www.python.org/ftp/python/${PYTHON_INSTALL_VERSION%%[a-z]*}/Python-$PYTHON_INSTALL_VERSION.tar.xz" \
+    && mkdir -p /usr/src/python \
+    && tar -xJC /usr/src/python --strip-components=1 -f python.tar.xz \
+    && rm python.tar.xz \
+    && cd /usr/src/python \
+    && gnuArch="$(dpkg-architecture --query DEB_BUILD_GNU_TYPE)" \
+    && ./configure \
+        --build="$gnuArch" \
+        --enable-loadable-sqlite-extensions \
+        --enable-optimizations \
+        --enable-option-checking=fatal \
+        --enable-shared \
+        --with-system-expat \
+        --with-system-ffi \
+        --without-ensurepip \
```

Comment:
@cnbennett3 probably need to remove this.

Reply:
Ah, yes--esp. the last one.

Reply:
Sorry, that's the only one I meant `--without-ensurepip`. I think we want the rest.